src/box/box.cc

```diff
@@ -3856,6 +3893,34 @@ bootstrap_journal_write(struct journal *base, struct journal_entry *entry)
 	return 0;
 }
 
+/**
+ * Wait until every remote peer that managed to connect chooses this node as its
+ * bootstrap leader and fail otherwise.
+ */
+static void
+check_bootstrap_unanimosity(void)
+{
+	replicaset_foreach(replica) {
+		struct applier *applier = replica->applier;
+		if (applier == NULL || applier->state != APPLIER_CONNECTED)
+			continue;
+		struct ballot *ballot = &applier->ballot;
+		const char *replica_str =
+			tt_sprintf("%s at %s", tt_uuid_str(&applier->uuid),
+				   applier_uri_str(applier));
+		say_info("Checking if %s chose this instance as bootstrap "
+			 "leader", replica_str);
+		if (tt_uuid_is_nil(&ballot->bootstrap_leader_uuid))
+			applier_wait_bootstrap_leader_uuid_is_set(applier);
```

Comment:
Can it be that on first wait this is fine, then we wait for another applier, and during that the first one changes its bootstrap leader choice?

Reply:
I thought about it for a moment, and I believe no.

Once a remote peer tells us its bootstrap leader choice, it stops trying to connect to anyone else. So its bootstrap leader is stabilised, because noone is booted yet and noone updates its ballot.
So even while the remote peer receives ER_LOADING (because the bootstrap leader still waits for others to vote for him), it will still try to boot from the same bootstrap leader over and over again until it finally succeeds.